### DISCUSSION TITLE
How to stop a client and start a new one?

### DISCUSSION THREAD

**Discussion by Prx001** [Rapptz/discord.py#6168] [Q&A] [answered]
Hi everyone!
I'm working on a program (Not developing a discord bot) which has 2 clients and a task loop for each of them, and one discord bot.
My whole entire program and codes run in these 2 clients, actually I used Discord API to help my program works well and communicate with it's extensions well.
It starts with a client and I want to execute second client if a certain conditions holds.
Here's my code:
```python
from discord.ext import tasks
from discord.ext.commands import Bot
import aiohttp.client_exceptions

TOKEN = "blablabla"

client = Bot(command_prefix=">")
@tasks.loop(minutes=10)
async def main0():
	# code
	some_condition = True
	await client.logout()                    #to end to the client after excution of all those code in main0 and execute the rest of clients in while loop
client1 = Bot(command_prefix=">")
@tasks.loop(minutes=10)
async def main1():
	# code
	some_condition = False
while True:
	try:
		main0.start()
		client.run(TOKEN)
		if some_condition == True:           #if some conditions happens:
			main1.start()
			client1.run(TOKEN)
		else:
			continue                         # to execute the first client again
	except aiohttp.client_exceptions.ClientConnectorError:
		print("You don't have internet access!")
```
As you can see, I used client.logout to get out of the first client; but it doesn't work properly and I think this is probably logout method problem not my code...
It stops doing anything after executes client.logout, I mean nothing happens and I also used breakpoints and PyCharm debug to see what happens, but nothing happens after client.logout, it keeps stuck there.
If is there any way to stop tasks.loop functions tell me please I'm a beginner here, I would use it and make one client with two functions instead of two clients and one function for each.
Any help will be appreciated.

**Comment by Rapptz** [comment, marked as answer]
`Client.run` takes ownership of the event loop and closes it when it terminates. If you want control over the event loop you'll have to use [`Client.login`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Client.login) with [`Client.connect`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Client.connect). For more information you can see [this issue comment](https://github.com/Rapptz/discord.py/issues/516#issuecomment-287511041).
